src/Microsoft.Data/PrimitiveColumn.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A column to hold primitive types such as int, float etc.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public partial class PrimitiveColumn<T> : BaseColumn
+        where T : unmanaged
+    {
+        private PrimitiveColumnContainer<T> _columnContainer;
+
+        internal PrimitiveColumn(string name, PrimitiveColumnContainer<T> column) : base(name, column.Length, typeof(T))
+        {
+            _columnContainer = column;
+        }
+
+        public PrimitiveColumn(string name, IEnumerable<T> values) : base(name, 0, typeof(T))
+        {
+            _columnContainer = new PrimitiveColumnContainer<T>(values);
+            Length = _columnContainer.Length;
+        }
+
+        public PrimitiveColumn(string name, long length = 0, bool isNullable = true) : base(name, length, typeof(T))
+        {
+            _columnContainer = new PrimitiveColumnContainer<T>(length);
+        }
+
+        public override object this[long startIndex, int length] {
```

Comment:
(nit) braces should go on a new line.

Reply:
Checked in an editorconfig file with the latest 2 commits. Shouldn't happen again